File: ts/src/short-name.ts
export function shortName(full: string): string {
  const hashIdx = full.indexOf("#");
  let base = hashIdx >= 0 ? full.substring(0, hashIdx) : full;
  const spec = hashIdx >= 0 ? full.substring(hashIdx) : "";
  const dotIdx = base.lastIndexOf(".");
  if (dotIdx >= 0) base = base.substring(dotIdx + 1);
  return base + spec;
}

File: py/src/rulecat/short_name.py
def short_name(full: str) -> str:
    hash_idx = full.find("#")
    if hash_idx >= 0:
        base = full[:hash_idx]
        spec = full[hash_idx:]
    else:
        base = full
        spec = ""
    dot_idx = base.rfind(".")
    if dot_idx >= 0:
        base = base[dot_idx + 1:]
    return base + spec
